staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/configmaplock.go

```diff
@@ -22,6 +22,7 @@ import (
 	"fmt"
 
 	"k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
```

Comment:
I'm wondering what the approach you use here means for clients that are not in k/k repo, but are vendoring this package.
I like the overall idea you did, but we need to clearly describe consequences:
- what are the next steps for us (to migrate things like scheduler, controller-manager to use just lease)
- what are the steps we suggest all clients vendoring that should do
- what changes we will do in this directory (and how it affects the above)

Reply:
There are two ideas in my mind about the next step.
- The first idea is to inject the lease into the resource lock without breaking the original interface, and the client does not perceive the process of switching. 
Switch it to lease on the next release, but worry that someone will upgrade the version that skips this migration (required in the release note)
- Another option is for the client to explicitly control the migration process through the flag.

I am implementing the former here, but I am not sure which one is better for us. We can continue to discuss this program in depth.

Reply:
> The first idea is to inject the lease into the resource lock without breaking the original interface, and the client does not perceive the process of switching.

That's not an option, because we don't control all users of the interface.

> Another option is for the client to explicitly control the migration process through the flag.

That sounds like the only reasonable one to me.